packages/iris-grid/src/IrisGridModelUpdater.tsx

```diff
@@ -71,6 +71,10 @@ const IrisGridModelUpdater = React.memo(
     columnHeaderGroups,
     partitionConfig,
   }: IrisGridModelUpdaterProps) => {
+    if (model.formatter !== formatter) {
```

Comment:
I think the "less" hacky way to fix this is to get rid of all the `useEffect`s in here. Those run after the render, when really all we want from this component is to set values when they have changed. And then make sure it checks on every render (i.e. the component shouldn't be memoized at the top), and then we don't need to pass in a `modelColumns` prop. Instead could make a hook to run immediately when something has changed, e.g.

```typescript
function useOnChange(callback: () => void, deps: DependencyList): void {
  // Want to call the callback if the deps have changed from the previous time
  const prevDeps = usePrevious(deps);
  if (prevDeps === undefined || !deps.every((dep, i) => dep === prevDeps[i])) {
    callback();
  }
}
```

Then instead of `useEffect` in `IrisGridModelUpdater`, just use `useOnChange`, and use `model.columns` instead of `modelColumns`, get rid of the `modelColumns` prop, and don't wrap the function in `React.memo`. I think that will work.